File: app/src/main/java/club/hutcwp/coolweather/adapter/CityListAdapter.java
/*
 * Copyright (c) 2017. The Android Project Created By Hutcwp.
 */

package club.hutcwp.coolweather.adapter;

import android.content.Context;
import android.support.annotation.LayoutRes;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.TextView;

import java.util.List;

import club.hutcwp.coolweather.R;

/**
 * Created by hutcwp on 2017/4/9.
 * Mail : hutcwp@foxmail.com
 * Blog : hutcwp.club
 * GitHub : github.com/hutcwp
 */

public class CityListAdapter extends ArrayAdapter<String> {


    private int resourceId;

    public CityListAdapter(@NonNull Context context, @LayoutRes int resource, List<String> list) {
        super(context, resource, list);
        resourceId = resource;

    }


    @NonNull
    @Override
    public View getView(int position, @Nullable View convertView, @NonNull ViewGroup parent) {

        String name = getItem(position);
        View view;
        ViewHolder viewHolder;
        if (convertView == null) {

            view = LayoutInflater.from(getContext()).inflate(resourceId, parent, false);

            viewHolder = new ViewHolder();
            viewHolder.textView = (TextView) view.findViewById(R.id.name);
            view.setTag(viewHolder);

        } else {
            view = convertView;
            viewHolder = (ViewHolder) view.getTag();

        }

        viewHolder.textView.setText(name);
        return view;
    }


    private class ViewHolder {

        TextView textView;

    }

}

File: app/src/main/java/club/hutcwp/coolweather/adapter/CityListAdapter.kt
/*
 * Copyright (c) 2017. The Android Project Created By Hutcwp.
 */

package club.hutcwp.coolweather.adapter

import android.content.Context
import android.support.annotation.LayoutRes
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import android.widget.TextView

import club.hutcwp.coolweather.R

/**
 * Created by hutcwp on 2017/4/9.
 * Mail : hutcwp@foxmail.com
 * Blog : hutcwp.club
 * GitHub : github.com/hutcwp
 */

class CityListAdapter(context: Context, @param:LayoutRes private val resourceId: Int, list: List<String>) : ArrayAdapter<String>(context, resourceId, list) {

    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
        val name = getItem(position)
        val view: View
        val viewHolder: ViewHolder
        if (convertView == null) {
            view = LayoutInflater.from(context).inflate(resourceId, parent, false)
            viewHolder = ViewHolder()
            viewHolder.textView = view.findViewById<View>(R.id.name) as TextView
            view.tag = viewHolder
        } else {
            view = convertView
            viewHolder = view.tag as ViewHolder
        }
        viewHolder.textView!!.text = name
        return view
    }

    private inner class ViewHolder {
        var textView: TextView? = null
    }
}
